pkg/reader/ccip.go

```diff
@@ -1146,117 +1148,76 @@ func (scc sourceChainConfig) check() (bool /* enabled */, error) {
+		return nil, fmt.Errorf("validate extended reader existence: %w", err)
 	}
 
-	res := make(map[cciptypes.ChainSelector]sourceChainConfig)
-	mu := new(sync.Mutex)
+	configs := make(map[cciptypes.ChainSelector]sourceChainConfig)
+	contractBatch := make(types.ContractBatch, 0, len(chains))
+	sourceChains := make([]any, 0, len(chains))
 
-	eg := new(errgroup.Group)
-	for _, chainSel := range chains {
-		if chainSel == r.destChain {
+	for _, chain := range chains {
+		if chain == r.destChain {
 			continue
 		}
+		sourceChains = append(sourceChains, chain)
 
-		// TODO: look into using BatchGetLatestValue instead to simplify concurrency?
-		eg.Go(func() error {
-			resp := sourceChainConfig{}
-			err := r.contractReaders[r.destChain].ExtendedGetLatestValue(
-				ctx,
-				consts.ContractNameOffRamp,
-				consts.MethodNameGetSourceChainConfig,
-				primitives.Unconfirmed,
-				map[string]any{
-					"sourceChainSelector": chainSel,
-				},
-				&resp,
-			)
-			if err != nil {
-				return fmt.Errorf("failed to get source chain config for source chain %d: %w",
-					chainSel, err)
-			}
-
-			enabled, err := resp.check()
-			if err != nil {
-				return fmt.Errorf("source chain config check for chain %d failed: %w", chainSel, err)
-			}
-			if !enabled {
-				// We don't want to process disabled chains prematurely.
-				r.lggr.Debugw("source chain is disabled", "chain", chainSel)
-				return nil
-			}
-
-			mu.Lock()
-			res[chainSel] = resp
-			mu.Unlock()
-			return nil
+		contractBatch = append(contractBatch, types.BatchRead{
+			ReadName: consts.MethodNameGetSourceChainConfig,
+			Params: map[string]any{
+				"sourceChainSelector": chain,
+			},
+			ReturnVal: new(sourceChainConfig),
 		})
 	}
 
-	if err := eg.Wait(); err != nil {
-		return nil, err
-	}
-	return res, nil
-}
-
-// selectorsAndConfigs wraps the return values from getAllSourceChainConfigs.
-type selectorsAndConfigs struct {
-	Selectors          []uint64            `mapstructure:"F0"`
-	SourceChainConfigs []sourceChainConfig `mapstructure:"F1"`
-}
-
-// getAllOffRampSourceChainsConfig get all enabled source chain configs from the offRamp for dest chain
-func (r *ccipChainReader) getAllOffRampSourceChainsConfig(
-	ctx context.Context,
-	lggr logger.Logger,
-) (map[cciptypes.ChainSelector]sourceChainConfig, error) {
-	if err := validateExtendedReaderExistence(r.contractReaders, r.destChain); err != nil {
-		return nil, fmt.Errorf("validate extended reader existence: %w", err)
-	}
-
-	configs := make(map[cciptypes.ChainSelector]sourceChainConfig)
-
-	var resp selectorsAndConfigs
-	err := r.contractReaders[r.destChain].ExtendedGetLatestValue(
-		ctx,
-		consts.ContractNameOffRamp,
-		consts.MethodNameOffRampGetAllSourceChainConfigs,
-		primitives.Unconfirmed,
-		map[string]any{},
-		&resp,
+	results, _, err := r.contractReaders[r.destChain].ExtendedBatchGetLatestValues(
+		ctx, contractreader.ExtendedBatchGetLatestValuesRequest{consts.ContractNameOffRamp: contractBatch},
+		false,
```

Comment:
Previously this was using an `Unconfirmed` confidence level. What level is used with batch get?

Reply:
It's ignored anyway on Solana. Looks like it does default to `Unconfirmed` on EVM, and the possibility of overriding the default looks mostly implemented. Not sure why they didn't expose that feature in the API:
https://github.com/smartcontractkit/chainlink/blob/develop/core/services/relay/evm/read/batch.go#L132-L140

https://github.com/smartcontractkit/chainlink/blob/develop/core/services/relay/evm/read/bindings.go#L177